docs/cloud-tutorials/best-practices.md

```diff
@@ -0,0 +1,91 @@
+* Your application is allocated 1GB of disk space, 1CPU and 512MB of RAM.
+
+### Deployment
+* An application must be registered before being deployed. DBOS Cloud CLI handles registration automatically when you deploy an application for the first time.
+* Interrupting a deployment request - for example, pressing control + C - will *not* cancel the deploy action in DBOS Cloud.
+* DBOS Cloud builds your application and will prune all development dependencies. Make sure to explicitly install any runtime dependency you need with `npm install`.
+* DBOS Cloud manages an internal version number for your application. You can download specific versions achives using the DBOS Cloud CLI. [TOLINK WHEN MERGED]
+
+### Runtime
+* Files on disk are ephemeral. Nothing on the filesystem is persisted between deployments.
+* To accomodate long running workflows, DBOS Cloud will keep around VMs running old versions of a workflow when you deploy a new version of the application. This also means you need to be mindful of non-backward compatible database schema changes, as they could break workflows running an older version of the code.
+
+### Deletion
+* Deleting an app is an asynchronous process and there is a small delay between a deletion request completing and the application being fully deleted. This means you could hit an "application is busy" error when re-deploying an application you just deleted.
+* An application will be deleted once all PENDING workflows -- across all versions -- are completed.
+* All application archives -- including previous versions -- will be deleted from DBOS Cloud's artifact store.
+* Applications are assigned a [dedicated database role during deployment](#database-roles). Deleting an application will delete this role.
+* You can elect to drop the application database alongside deleting an application. This will also delete all the [database roles DBOS Cloud manages on behalf of the application](#database-roles).
+
+### Time travel
+Time travel can be enabled when deploying an application for the first time. Existing applications will need to be deleted and re-deployed to enable time travel.
+See more information in the [time travel](#time-travel) section.
+
+### Swapping database instance
+While an application cannot change database once registered, you can change the database instance (server) hosting the application database. When doing so, you can also decide to redeploy a previous version of the application. This is how DBOS Cloud support [point in time recovery](database-management#database-recovery) today.
+
+## Application databases
+
+:::tip
+Application databases are databases within a Postgres database instance (server).
+:::
+
+* Application databases cannot be shared across applications.
+* An application cannot change its database once deployed. You can delete the application and re-deploy with a new database name.
+
+### Database roles
+* DBOS Cloud will create the following roles for each application:
```

Comment:
I don't know about this section. We don't list the privileges of dbosadmin but we list the privileges of `dbos_role` in great detail. That seems inconsistent. That aside, why are we listing those things to begin with? This does not seem like a "best practice" but more of a "explanation how the roles work" Fair and maybe useful but what do we want the user to do about it? How should the user use these roles and for what?